libraries/scripting/dependencies/src/kotlin/script/experimental/dependencies/resolverNamedOptions.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package kotlin.script.experimental.dependencies
+
+fun ExternalDependenciesResolver.Options.value(name: DependenciesResolverOptionsName) =
+    value(name.key)
+
+fun ExternalDependenciesResolver.Options.flag(name: DependenciesResolverOptionsName) =
+    flag(name.key)
+
+operator fun MutableMap<String, String>.set(key: DependenciesResolverOptionsName, value: String) {
+    put(key.key, value)
+}
+
+
+enum class DependenciesResolverOptionsName {
```

Comment:
These options seem very specific to the individual dependencies resolvers. Perhaps it would be more appropriate to prefix this with a `Maven*` or `Ivy*` respectively.

This API is not implemented by many other resolvers that I know of. But what I want to avoid is someone implementing a dependencies resolver and thinking they:
1. have to support these exact options
1. are only allowed to support these exact options

@ligee what do you think?

Reply:
Thanks, @nerdsupremacist, I think the concern is valid.
But we can probably get away right now simply by moving it into the `impl` package, signalling that it is a mere implementation detail now.
@ileasile, what do you think about this?

Reply:
I see these options names only as a convenient and safe way to access "text" properties. Maybe it's worth adding somewhere in comments that these options are not mandatory? Also, I'm OK with moving it into the `impl` package.

Reply:
I understand your intention here, but we don't want to expose it as a generic API. So, please move it to the `impl`.

Reply:
Sure. Done ✔️